src/com/ppm/integration/agilesdk/connector/versionone/model/VersionOneTimebox.java

```diff
@@ -0,0 +1,148 @@
+	private String timeboxName;
+	private String beginDate;
+	private String endDate;
+	private String stateCode;
+	private List<VersionOneStory> stories;
+
+	public VersionOneTimebox(String timeboxName, String beginDate, String endDate, String stateCode) {
+
+		this.timeboxName = timeboxName;
+		this.beginDate = beginDate;
+		this.endDate = endDate;
+		this.stateCode = stateCode;
+	}
+
+	public String getTimeboxName() {
+		return timeboxName;
+	}
+
+	public void setTimeboxName(String timeboxName) {
+		this.timeboxName = timeboxName;
+	}
+
+	public String getBeginDate() {
+		return beginDate;
+	}
+
+	public void setBeginDate(String beginDate) {
+		this.beginDate = beginDate;
+	}
+
+	public String getEndDate() {
+		return endDate;
+	}
+
+	public void setEndDate(String endDate) {
+		this.endDate = endDate;
+	}
+
+	public String getStateCode() {
+		return stateCode;
+	}
+
+	public void setStateCode(String stateCode) {
+		this.stateCode = stateCode;
+	}
+
+	public List<VersionOneStory> getStories() {
+		return stories;
+	}
+
+	public void setStories(List<VersionOneStory> stories) {
+		this.stories = stories;
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((beginDate == null) ? 0 : beginDate.hashCode());
+		result = prime * result + ((endDate == null) ? 0 : endDate.hashCode());
+		result = prime * result + ((timeboxName == null) ? 0 : timeboxName.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		VersionOneTimebox other = (VersionOneTimebox) obj;
+		if (beginDate == null) {
+			if (other.beginDate != null)
+				return false;
+		} else if (!beginDate.equals(other.beginDate))
+			return false;
+		if (endDate == null) {
+			if (other.endDate != null)
+				return false;
+		} else if (!endDate.equals(other.endDate))
+			return false;
+		if (timeboxName == null) {
+			if (other.timeboxName != null)
+				return false;
+		} else if (!timeboxName.equals(other.timeboxName))
+			return false;
+		return true;
+	}
+
+	@Override
+	public String getName() {
+
+		return this.timeboxName;
+	}
+
+	@Override
+	public Date getScheduleFinish() {
+
+		return toDate(this.endDate);
+	}
+
+	@Override
+	public Date getScheduleStart() {
+
+		return toDate(this.beginDate);
+	}
+
+	@Override
+	public TaskStatus getStatus() {
+
+		return getTaskStatus(this.stateCode);
+	}
+
+	@Override
+	public List<IExternalTask> getChildren() {
+		List<IExternalTask> list = new ArrayList<>();
+		for (VersionOneStory s : stories) {
+			list.add(s);
+		}
+
+		return list;
+	}
+
+	private TaskStatus getTaskStatus(String status) {
+
+		switch (status) {
+		case "CLSD":
+			return TaskStatus.COMPLETED;
+		case "FUTR":
+			return TaskStatus.IN_PLANNING;
+		case "ACTV":
+			return TaskStatus.ACTIVE;
+		default:
+			return TaskStatus.UNKNOWN;
```

Comment:
Can this case happen, or is it more like a error handler? Be careful when using status UNKNOWN - you should NOT use it in normal business operation.